src/core/env/ClientContext.tsx

```diff
@@ -38,11 +40,19 @@ type ClientContextProps = {
 const ClientContext: FC<ClientContextProps> = ({
   children,
   envVars,
+  headers,
   lang,
   messages,
   user,
 }) => {
-  const env = new Environment(new BrowserApiClient(), envVars);
+  const onServer = typeof window == 'undefined';
+
+  const apiClient = onServer
+    ? new BackendApiClient(headers)
+    : new BrowserApiClient();
+
+  const env = new Environment(apiClient, envVars);
```

Comment:
This fixes rendering on Vercel. The backend needs to have a proper API client in order to make requests for data, or it will get errors when making the requests on the server, causing HTTP status 500 errors.

In other places I often use the `useServerSide()`, which is a hook that sets a state to `true` and then resets it to false after the first client-side render using `useEffect()`. But that doesn't work in this context, because it only updates _after the first render_. The first render happens with the state as `true` and in this case the result would be that the first render on the client would trigger API requests using the backend API client.